fix(animations): remove `finish` listener once player is destroyed

arturovt opened the pull request:
This commit removes the `finish` listener from the Animation object once the animation is finished, effectively resolving a memory leak. Previously, the `finish` listener captured `this`, which prevented `this` from being garbage collected.

See the image below:

![Screenshot from 2023-07-22 00-13-37](https://github.com/angular/angular/assets/7337691/2a800642-95bf-4f39-a8b7-3de06b10444b)

JeanMeche reviewed (approved):
LGTM. 

Depending on which one will get merged first between this one and  #50662. We'll need a rebase;

arturovt commented:
@JeanMeche sure, ping me back if that one you mentioned will be merged first. Because I may not be notified 😁

arturovt committed:
fix(animations): remove `finish` listener once player is destroyed

This commit removes the `finish` listener from the Animation object once
the animation is finished, effectively resolving a memory leak. Previously,
the `finish` listener captured `this`, which prevented `this` from being garbage collected.

arturovt commented:
@JeanMeche I saw the PR you mentioned was merged, can we also progress on this one (at least trigger TGP to see whether i's failing or not)?

alxhub commented:
This PR was merged into the repository by commit a02a745a4ada46a084235f94b68d5defd224ddcf.
